docarray/document/mixins/porting.py

```diff
@@ -80,12 +83,15 @@ def to_dict(self, protocol: str = 'jsonschema', **kwargs) -> Dict[str, Any]:
                 self.to_protobuf(),
                 **kwargs,
             )
-        else:
+        elif protocol == 'dynamic':
             warnings.warn(
-                f'protocol=`{protocol}` is not supported, '
-                f'the result dict is a Python dynamic typing dict without any promise on the schema.'
+                'The result dict is a Python dynamic typing dict without any promise on the schema.'
             )
-            return dataclasses.asdict(self._data)
+            res = dataclasses.asdict(self._data)
+            del res['_reference_doc']
+            return res
```

Comment:
no need, you only need it in `load/from`

we do not support arbitrary/unschema-ed json dumping